packages/zoe/src/contractFacet/contractFacet.js

```diff
@@ -0,0 +1,348 @@
+
+        seatStagings.forEach(seatStaging =>
+          assert(
+            allSeatStagings.has(seatStaging),
+            details`The seatStaging ${seatStaging} was not recognized`,
+          ),
+        );
+
+        // Ensure that rights are conserved overall. Offer safety was
+        // already checked when an allocation was staged for an individual seat.
+        const flattened = arr => [].concat(...arr);
+        const flattenAllocations = allocations =>
+          flattened(allocations.map(allocation => Object.values(allocation)));
+
+        const previousAllocations = seatStagings.map(seatStaging =>
+          seatStaging.getSeat().getCurrentAllocation(),
+        );
+        const previousAmounts = flattenAllocations(previousAllocations);
+
+        const newAllocations = seatStagings.map(seatStaging =>
+          seatStaging.getStagedAllocation(),
+        );
+        const newAmounts = flattenAllocations(newAllocations);
+
+        assert(
+          areRightsConserved(getAmountMath, previousAmounts, newAmounts),
+          details`Rights are not conserved in the proposed reallocation`,
+        );
+
+        // Commit the staged allocations and inform Zoe of the
+        // newAllocation.
+        seatStagings.forEach(seatStaging =>
+          seatToZCFSeatAdmin.get(seatStaging.getSeat()).commit(seatStaging),
+        );
+      },
+      assertUniqueKeyword: keyword => {
+        assertKeywordName(keyword);
+        assert(
+          !getKeywords(instanceRecord.issuerKeywordRecord).includes(keyword),
+          details`keyword ${keyword} must be unique`,
+        );
+      },
+      saveIssuer: (issuerP, keyword) => {
+        // TODO: The checks of the keyword for uniqueness are
+        // duplicated. Assess how waiting on promises to resolve might
+        // affect those checks and see if one can be removed.
+        zcf.assertUniqueKeyword(keyword);
+        return E(zoeInstanceAdmin)
+          .saveIssuer(issuerP, keyword)
+          .then(() => {
+            return issuerTable
+              .getPromiseForIssuerRecord(issuerP)
+              .then(record => registerIssuerRecord(keyword, record));
+          });
+      },
+      makeInvitation: (offerHandler, description, customProperties = {}) => {
+        assert.typeof(
+          description,
+          'string',
+          details`invitations must have a description string: ${description}`,
+        );
+
+        const invitationHandle = /** @type {InvitationHandle} */ (harden({}));
+        invitationHandleToHandler.init(invitationHandle, offerHandler);
+        /** @type {Promise<Payment>} */
+        const invitationP = E(zoeInstanceAdmin).makeInvitation(
+          invitationHandle,
+          description,
+          customProperties,
+        );
+        return invitationP;
+      },
+      // Shutdown the entire vat and give payouts
+      shutdown: () => E(zoeInstanceAdmin).shutdown(),
+
+      // The methods below are pure and have no side-effects //
```

Comment:
OOPS! @Chris-Hibbert and I both broke this comment when we added `makeZCFMint` and `makeEmptySeatKit` methods to the block below this comment. The comment is good. Having such a split is good. Please reorder the methods. Also reorder the declaration in types.js so they continue to match. Please repeat this comment between the two groups of methods there.

Reply:
Let's tackle the reordering in type.js later (I certainly was not doing any particular ordering there!)